fuse_models/src/unicycle_2d.cpp

```diff
@@ -348,8 +352,20 @@ void Unicycle2D::generateMotionModel(
   state_history_.emplace(beginning_stamp, std::move(state1));
   state_history_.emplace(ending_stamp, std::move(state2));
 
-  // Scale process noise covariance pose by the norm of the current state twist
+
   auto process_noise_covariance = process_noise_covariance_;
+
+  // Rotate the process noise covariance with the yaw angle of the current state orientation
+  if (rotate_process_noise_covariance_to_state_orientation_)
+  {
+    auto rotation_matrix = Eigen::Rotation2Dd(state1.pose.yaw()).toRotationMatrix();
```

Comment:
This can be `const`:
```suggestion
    const auto rotation_matrix = Eigen::Rotation2Dd(state1.pose.yaw()).toRotationMatrix();
```